### PR TITLE
Adding new transition scenarios to the SDK samples

### PR DESCRIPTION
**Pull request by cbarneyMS**
# Description

Adding SDK samples for TransitionSubscription and GetSubscriptionTransitions.

## Related issue

Kindly link any related issues (e.g. Fixes #xyz).

### CODE REVIEW

File: sdk/SdkSamples/Subscriptions/GetSubscriptionTransitions.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
+﻿// -----------------------------------------------------------------------
+// <copyright file="GetSubscriptionTransitions.cs" company="Microsoft">
+//      Copyright (c) Microsoft Corporation.  All rights reserved.
+// </copyright>
+// -----------------------------------------------------------------------
+
+namespace Microsoft.Store.PartnerCenter.Samples.Subscriptions
+{
+    /// <summary>
+    /// A scenario that retrieves a customer subscription.
```
Comment (brijeshp-ms) [inline sdk/SdkSamples/Subscriptions/GetSubscriptionTransitions.cs:10]: customer subscription transitions

File: sdk/SdkSamples/Subscriptions/TransitionSubscription.cs [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+// </copyright>
+// -----------------------------------------------------------------------
+
+namespace Microsoft.Store.PartnerCenter.Samples.Subscriptions
+{
+    using Microsoft.Store.PartnerCenter.Models.Subscriptions;
+    using System.Linq;
+
+    /// <summary>
+    /// A scenario that upgrades a customer subscription to a higher skew.
```
Comment (brijeshp-ms) [inline sdk/SdkSamples/Subscriptions/TransitionSubscription.cs:13]: I think this summary needs change as per transition

**Review by brijeshp-ms — APPROVED** [review]

File: sdk/SdkSamples/Subscriptions/GetSubscriptionTransitions.cs [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        /// Executes the scenario.
+        /// </summary>
+        protected override void RunScenario()
+        {
+            var partnerOperations = this.Context.UserPartnerOperations;
+            string customerId = this.ObtainCustomerId("Enter the ID of the customer whom to retrieve their Subscriptions");
+            string subscriptionId = this.ObtainSubscriptionId(customerId, "Enter the ID of the subscription to find transtions for");
+            var subscriptionOperations = partnerOperations.Customers.ById(customerId).Subscriptions.ById(subscriptionId);
+
+            this.Context.ConsoleHelper.StartProgress("Retrieving customer subscription Transitions");
```
Comment (kaminasy) [inline sdk/SdkSamples/Subscriptions/GetSubscriptionTransitions.cs:32]: nit: make it lowercase

File: sdk/SdkSamples/Subscriptions/GetSubscriptionTransitions.cs [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+            var transitions = subscriptionOperations.Transitions.Get();
+            this.Context.ConsoleHelper.StopProgress();
+
+            this.Context.ConsoleHelper.WriteObject(transitions, "Customer Subscription Transitions");
```
Comment (kaminasy) [inline sdk/SdkSamples/Subscriptions/GetSubscriptionTransitions.cs:36]: nit: make it lowercase

**Review by kaminasy — CHANGES_REQUESTED** [review]

File: sdk/SdkSamples/Subscriptions/GetSubscriptionTransitions.cs (hunk 1 above)
Comment (cbarneyMS) [reply sdk/SdkSamples/Subscriptions/GetSubscriptionTransitions.cs]: Updated

File: sdk/SdkSamples/Subscriptions/GetSubscriptionTransitions.cs (hunk 3 above)
Comment (cbarneyMS) [reply sdk/SdkSamples/Subscriptions/GetSubscriptionTransitions.cs]: Updated

File: sdk/SdkSamples/Subscriptions/TransitionSubscription.cs (hunk 2 above)
Comment (cbarneyMS) [reply sdk/SdkSamples/Subscriptions/TransitionSubscription.cs]: Updated

File: sdk/SdkSamples/Subscriptions/GetSubscriptionTransitions.cs (hunk 4 above)
Comment (cbarneyMS) [reply sdk/SdkSamples/Subscriptions/GetSubscriptionTransitions.cs]: Updated

**Review by kaminasy — APPROVED** [review]

**Pull request merged**